src/ILLink.Shared/DiagCode.cs

```diff
@@ -0,0 +1,13 @@
+
+namespace ILLink.Shared
+{
+	internal enum DiagCode
+	{
+		WRN_UsingRequiresUnreferencedCode = 2026
```

Comment:
Just double-checking, the point would be to use this enum even if the analyzer doesn't use the id, meaning it doesn't have to be shared between linker and analyzer to be in the enum and therefore we can migrate whichever id from error-codes here

Reply:
There was a discussion about having an enum for error codes in the past and it was deliberately not done: https://github.com/mono/linker/issues/986#issuecomment-610228419.

Reply:
I think that discussion is a bit obsolete now because the linker repo is no longer where we want to document error codes, so whatever pattern we choose in code should be purely for our benefit. The documentation for warnings should be on docs.microsoft.com (I'll try to start moving stuff over shortly).

This PR is still in draft because I haven't quite figured out the most useful pattern here.